### PR TITLE
CreditTags Reimplementation.

### PR DESCRIPTION
**Pull request by Jesus-QC**
Reimplemented credit tags. It now uses https://github.com/Exiled-Official/CreditTags as its database.

### CODE REVIEW

File: EXILED/Exiled.CreditTags/Commands/ShowCreditTag.cs [hunk 1]
```diff
@@ -36,18 +36,14 @@ public bool Execute(ArraySegment<string> arguments, ICommandSender sender, out s
         {
             CommandSender cmdSender = (CommandSender)sender;
 
-            if (!(Player.Get(cmdSender.SenderId) is Player player))
+            if (Player.Get(cmdSender.SenderId) is not { } player)
```
Comment (louis1706) [inline EXILED/Exiled.CreditTags/Commands/ShowCreditTag.cs:39]:
```suggestion
            if (Player.Get(cmdSender.SenderId) is not Player player)
```

File: EXILED/Exiled.CreditTags/CreditTags.cs [hunk 2]
```diff
@@ -134,28 +88,28 @@ void ShowRank(RankType rank)
...
+                            SetCreditCustomInfo(player, value);
+
+                        break;
+                    case InfoSide.FirstAvailable:
+                        if (canReceiveCreditBadge)
+                            SetCreditBadge(player, value);
+                        else if (canReceiveCreditCustomInfo)
+                            SetCreditCustomInfo(player, value);
+
+                        break;
```
Comment (louis1706) [inline EXILED/Exiled.CreditTags/CreditTags.cs:112]: can you used C# 9 implementation for switch on this one ?
with used of 
switch x when y

File: EXILED/Exiled.CreditTags/Events/CreditsHandler.cs [hunk 3]
```diff
@@ -24,7 +24,8 @@ internal sealed class CreditsHandler
         /// <param name="ev"><inheritdoc cref="VerifiedEventArgs"/></param>
         public void OnPlayerVerify(VerifiedEventArgs ev)
         {
-            Timing.CallDelayed(0.5f, () => Instance.ShowCreditTag(ev.Player, null, null));
+            DatabaseHandler.UpdateData();
+            Timing.CallDelayed(0.5f, () => Instance.ShowCreditTag(ev.Player));
```
Comment (louis1706) [inline EXILED/Exiled.CreditTags/Events/CreditsHandler.cs:28]: no more need to delay this stuff now or only one tick is enough

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: EXILED/Exiled.CreditTags/Features/DatabaseHandler.cs [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                    ETagCache = File.ReadAllText(ETagCachePath);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error($"{nameof(DatabaseHandler)}: There was an error reading the cache files.");
+                Log.Error(e);
+            }
+        }
+
```
Comment (Vladislav-CS) [inline EXILED/Exiled.CreditTags/Features/DatabaseHandler.cs:54]: I don't like the thing it writes caches in the file, why we can't keep it in memory?

File: EXILED/Exiled.CreditTags/Events/CreditsHandler.cs (hunk 3 above)
Comment (Misfiy) [reply EXILED/Exiled.CreditTags/Events/CreditsHandler.cs:28]: Delay is required

**Review by louis1706 — APPROVED** [review]

**Review by Misfiy — APPROVED** [review]

**Pull request merged**